packages/create-toolpad-app/index.ts

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env node
+
+import chalk from 'chalk';
+import { execaCommand } from 'execa';
+import inquirer from 'inquirer';
+import fs from 'fs';
+import { getPackageManager } from './utils/getPackageManager';
+
+// Detect the package manager
+const packageManager = getPackageManager();
+
+// Create a new directory and initialize a new project
+const scaffoldProject = async () => {
+  const name = await inquirer.prompt([
+    {
+      type: 'input',
+      name: 'projectName',
+      message: 'Enter the name of your project:',
+      default: 'my-app',
+    },
+  ]);
+
+  // eslint-disable-next-line no-console
+  console.log(`Creating a new MUI Toolpad project in ${chalk.blue(name.projectName)}`);
+  try {
+    fs.mkdirSync(name.projectName);
+    process.chdir(name.projectName);
+
+    const packageJson = {
+      name: name.projectName,
+      version: '0.1.0',
+      private: true,
+      scripts: {
+        dev: 'toolpad dev',
+        build: 'toolpad build',
+        start: 'toolpad start',
+      },
+    };
+
+    fs.writeFileSync('package.json', JSON.stringify(packageJson, null, 2));
+    return name.projectName;
+  } catch (error) {
+    if (fs.existsSync(name.projectName)) {
+      console.error(
+        `Directory ${chalk.red(name.projectName)} already exists. Please provide a different name.`,
+      );
+    }
+    return null;
+  }
+};
+
+// Install the dependencies
+const installDeps = async () => {
+  const installVerb = packageManager === 'yarn' ? 'add' : 'install';
+  const command = `${packageManager} ${installVerb} @mui/toolpad @mui/toolpad-core`;
+  await execaCommand(command, { stdio: 'inherit' });
+
+  // eslint-disable-next-line no-console
+  console.log(`${chalk.green('success')} - Dependencies installed successfully!`);
+};
+
+// Run the CLI interaction with Inquirer.js
+const run = async () => {
+  let projectName;
+  do {
+    // eslint-disable-next-line no-await-in-loop
+    projectName = await scaffoldProject();
+  } while (!projectName);
+
+  const installDependenciesConsent = await inquirer.prompt([
```

Comment:
I feel like since we're creating a new app from scratch, and not changing an existing one, that we wouldn't need to ask for this consent? But maybe I'm missing something.

Reply:
I thought it was a good to have in a CLI before installing packages in the user's system, to sort of give a final heads-up before going for this bandwidth heavy operation

Reply:
It's fine if you want to keep it, this is just my opinion anyway, I'm not super sure about it :)

Reply:
@bharatkashyap can you check with a few similar cli tools to see what the usual workflow is?

Reply:
Create Next App doesn't have something like this, it just lists the dependencies and proceeds to install them. 

<img width="705" alt="Screenshot 2023-03-07 at 3 24 42 PM" src="https://user-images.githubusercontent.com/19550456/223450658-8ac14896-51f9-405f-8e21-122c0d1a170a.png">

It makes sense since the operation may be cancelled if the user doesn't intend to instal deps

Reply:
Yeah, if no other CLI tools are doing I think we could definitely skip this prompt.

Reply:
Is there value in prompting if the install would happen in an existing non-empty folder?

Reply:
> Is there value in prompting if the install would happen in an existing non-empty folder?

That isn't possible in the current shape, since we try and create a new directory and fail if the directory name provided exists